shell/browser/native_window_mac.mm

```diff
@@ -930,11 +933,18 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
 }
 
 void NativeWindowMac::SetMaximizable(bool maximizable) {
-  [[window_ standardWindowButton:NSWindowZoomButton] setEnabled:maximizable];
+  if (maximizable_ != maximizable) {
+    maximizable_ = maximizable;
+    [[window_ standardWindowButton:NSWindowZoomButton] setEnabled:maximizable];
+  }
```

Comment:
With this new revision, [setEnabled] is not reached in this call from the ctor if the user specified a `maximizable` as `true` (because `maximizable_` is already set to `true` via the header declaration):

```
    // Set maximizable state last to ensure zoom button does not get reset
    // by calls to other APIs.
    SetMaximizable(maximizable);
```

I'm 90% sure this is safe but wanted to ask to confrm -- when a new window is created on macOS, is its NSWindowZoomButton is enabled by default?

Reply:
yes - it's default true! frameless windows and a few other things mean it's false, but there's no case where it would be false where it's not specifically _set_ to false in a fn or ctor!